fast_api_als/routers/three_pl.py

```diff
@@ -10,17 +10,26 @@
 
 router = APIRouter()
 
+log = logging.getLogger(__name__)
 
 @router.post("/reset_authkey")
 async def reset_authkey(request: Request, token: str = Depends(get_token)):
-    body = await request.body()
-    body = json.loads(body)
+    try:
+        body = await request.body()
+        body = json.loads(body)
+        log.info("Body parsed successfully.")
+    except json.JSONDecodeError as jde : 
+        log.error(f'Request body not a valid JSON document: {jde}')
+    except Exception as e:
+        log.error(f'Request body parse failed: {e}')
     provider, role = get_user_role(token)
     if role != "ADMIN" and (role != "3PL"):
+        raise HTTPException(status_code=401, detail='User with role other than ADMIN or 3PL are not authorized.')
         pass
```

Comment:
remove line 28, log the error. Line 53 too

Reply:
Fixed.